tests/dataset_api_tests.py

```diff
@@ -0,0 +1,445 @@
+        }
+        uri = "api/v1/dataset/"
+        rv = self.client.post(uri, json=table_data)
+        self.assertEqual(rv.status_code, 201)
+        data = json.loads(rv.data.decode("utf-8"))
+        model = db.session.query(SqlaTable).get(data.get("id"))
+        self.assertIn(admin, model.owners)
+        self.assertIn(alpha, model.owners)
+        db.session.delete(model)
+        db.session.commit()
+
+    def test_create_dataset_item_owners_invalid(self):
+        """
+            Dataset API: Test create dataset item owner invalid
+        """
+        admin = self.get_user("admin")
+        example_db = get_example_database()
+        self.login(username="admin")
+        table_data = {
+            "database": example_db.id,
+            "schema": "",
+            "table_name": "ab_permission",
+            "owners": [admin.id, 1000],
+        }
+        uri = f"api/v1/dataset/"
+        rv = self.client.post(uri, json=table_data)
+        self.assertEqual(rv.status_code, 422)
+        data = json.loads(rv.data.decode("utf-8"))
+        expected_result = {"message": {"owners": ["Owners are invalid"]}}
+        self.assertEqual(data, expected_result)
+
+    def test_create_dataset_validate_uniqueness(self):
+        """
+            Dataset API: Test create dataset validate table uniqueness
+        """
+        example_db = get_example_database()
+        self.login(username="admin")
+        table_data = {
+            "database": example_db.id,
+            "schema": "",
+            "table_name": "birth_names",
+        }
+        uri = "api/v1/dataset/"
+        rv = self.client.post(uri, json=table_data)
+        self.assertEqual(rv.status_code, 422)
+        data = json.loads(rv.data.decode("utf-8"))
+        self.assertEqual(
+            data, {"message": {"table_name": ["Datasource birth_names already exists"]}}
+        )
+
+    def test_create_dataset_validate_database(self):
+        """
+            Dataset API: Test create dataset validate database exists
+        """
+        self.login(username="admin")
+        table_data = {"database": 1000, "schema": "", "table_name": "birth_names"}
+        uri = "api/v1/dataset/"
+        rv = self.client.post(uri, json=table_data)
+        self.assertEqual(rv.status_code, 422)
+        data = json.loads(rv.data.decode("utf-8"))
+        self.assertEqual(data, {"message": {"database": ["Database does not exist"]}})
+
+    def test_create_dataset_validate_tables_exists(self):
+        """
+            Dataset API: Test create dataset validate table exists
+        """
+        example_db = get_example_database()
+        self.login(username="admin")
+        table_data = {
+            "database": example_db.id,
+            "schema": "",
+            "table_name": "does_not_exist",
+        }
+        uri = "api/v1/dataset/"
+        rv = self.client.post(uri, json=table_data)
+        self.assertEqual(rv.status_code, 422)
+
+    @patch("superset.datasets.dao.DatasetDAO.create")
```

Comment:
I think this test might be implemented at the wrong level - we're patching an item 2 levels deep (view -> command -> DAO) - it might be preferable to have 2 tests, one checking that the Command throws the proper error in the situation, and another ensuring that the error is handled correctly.

Reply:
I'm ensuring that low level (like backend) errors are handled correctly by the API, and get really high coverage levels.